extensions/wikia/Email/Controller/ForgotPasswordController.class.php

```diff
@@ -15,74 +14,54 @@
-		);
-	}
+	const TRACKING_CATEGORY = TrackingCategories::FORGOT_PASSWORD;
 
-	public function assertCanEmail() {
-		parent::assertCanEmail();
+	private $tempPass;
 
-		$this->assertCanChangePassword();
-		$this->assertPasswordReminderNotThrottled();
-		$this->assertHasIP();
+	public function initEmail() {
+		$userService = new \UserService();
+		$this->tempPass = $this->request->getVal(
+			'tempPass',
+			$userService->resetPassword( $this->targetUser )
+		);
 	}
 
 	public function getSubject() {
-		return $this->getMessage( 'emailext-password-email-subject' )->text();
-	}
-
-	protected function getContent() {
-		$html = $this->app->renderView(
-			get_class( $this ),
-			'body',
-			$this->request->getParams()
-		);
-
-		return $html;
+		return $this->getMessage( 'emailext-password-subject' )->text();
 	}
 
 	/**
 	 * @template forgotPassword
-	 *
-	 * @throws \MWException
 	 */
 	public function body() {
-		$targetUser = $this->targetUser;
-
-		wfRunHooks( 'User::mailPasswordInternal', [
-			$this->currentUser,
-			$this->getContext()->getRequest()->getIP(),
-			$targetUser,
-		] );
-
-		$tempPass = $targetUser->randomPassword();
-		$targetUser->setNewpassword( $tempPass );
-		$targetUser->saveSettings();
-
 		$this->response->setData( [
-			'greeting' => $this->getMessage( 'emailext-password-email-greeting', $targetUser->getName() )->text(),
-			'content' => $this->getMessage( 'emailext-password-email-content', $tempPass )->text(),
-			'signature' => $this->getMessage( 'emailext-password-email-signature' )->text(),
+			'salutation' => $this->getSalutation(),
+			'summary' => $this->getSummary(),
+			'passwordIntro' => $this->getIntro(),
+			'tempPassword' => $this->tempPass,
+			'unrequested' => $this->getMessage( 'emailext-password-unrequested' )->text(),
+			'questions' => $this->getMessage( 'emailext-password-questions' )->parse(),
+			'signature' => $this->getMessage( 'emailext-password-signature' )->text(),
 		] );
 	}
 
-	protected function assertCanChangePassword() {
-		if ( !$this->wg->Auth->allowPasswordChange() ) {
-			throw new Fatal( 'This user is not allowed to change their password' );
-		}
+	protected function getSummary() {
+		return $this->getMessage( 'emailext-password-summary' )->text();
+	}
+
+	protected function getIntro() {
+		return $this->getMessage( 'emailext-password-intro' )->text();
 	}
 
-	protected function assertPasswordReminderNotThrottled() {
-		// Do not throttle staff
-		if ( $this->wg->User->isStaff() ) {
-			return;
-		}
+	protected function getDetails() {
+		return $this->getMessage( 'emailext-password-details' )->text();
+	}
 
-		if ( $this->targetUser->isPasswordReminderThrottled() ) {
-			throw new Check( 'Too many resend password requests sent' );
-		}
+	protected function getContentFooterMessages() {
+		$contactUrl = '';
+		return [
+			$this->getMessage( 'emailext-password-footer-1', $contactUrl )->parse(),
+			$this->getMessage( 'emailext-password-footer-2' )->escaped(),
+		];
 	}
 }
```

Comment:
What problems would it cause?  It will only be used internally by dev, QA and LOC.